### PR TITLE
dumpling: fix consistency lock table when tables to dump is empty

### PR DESCRIPTION
**Pull request by lichunzhu**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/38683

Problem Summary:

### What is changed and how it works?
Retract consistency lock to consistency none once we find `len(lockTableSQL) == 0`. Including close connection for consistency lock and always `return true` when we try to Ping this connection.

### Check List

Tests 

- [x] Unit test
- [x] Integration test

Side effects

- None

Documentation

- None

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by D3Hunter — APPROVED** [review]
lgtm

File: dumpling/export/consistency_test.go [hunk 1]
```diff
@@ -107,6 +107,39 @@ func TestConsistencyLockControllerRetry(t *testing.T) {
...
+	_, ok := ctrl.(*ConsistencyLockDumpingTables)
+	require.True(t, ok)
+	require.NoError(t, ctrl.Setup(tctx))
+	require.NoError(t, ctrl.TearDown(tctx))
+
+	// should remove table db1.t1 in tables to dump
+	expectedDumpTables := NewDatabaseTables().
+		AppendViews("db2", "t4")
+	expectedDumpTables["db1"] = make([]*TableInfo, 0)
+	require.Equal(t, expectedDumpTables, conf.Tables)
```
Comment (lance6716) [inline dumpling/export/consistency_test.go:139]: we check `ctrl.Setup` for this no-table case?

**Review by lance6716 — APPROVED** [review]
rest lgtm

File: dumpling/export/consistency_test.go (hunk 1 above)
Comment (lichunzhu) [reply dumpling/export/consistency_test.go:139]: This check includes `no-table` case. After `db1`.`tb1` is removed there is no more table.

**Pull request merged**